### ISSUE TITLE
[SPEC] Distribution-Shifting Reference Card — Sibling to Dark Prose

### ISSUE THREAD

**Issue by michael-conrad** [michael-conrad/.opencode#848]
### Prerequisites

- None (this is the root dependency)

### Problem

The Dark Prose Reference Card (`guidelines/250-dark-prose-reference.md`) defines identity-framing patterns that shape **what** AI-agent-facing text says. Dark prose operates on the **content layer**.

Research on LLM output distributions reveals that content alone is necessary but not sufficient. A model can produce the *mean version* of any dark prose pattern — something that *looks* professional but contains hedging, vague language, and consensus-seeking.

The current system has no mechanism for addressing the **distribution layer** — how prompt, persona, and instruction structure causes the model to sample from different regions of its training data. Dark prose says *what to say*. Distribution shifting says *how to structure what you say so the model actually produces it instead of the mean equivalent*.

Research basis:

(1) **RLHF reduces output diversity** — Kirk et al. (2024, ICLR, https://arxiv.org/abs/2310.06452) find RLHF produces significantly less diverse outputs than SFT across lexical, semantic, and perspective measures. This confirms RLHF narrows the output distribution toward a mean response.

(2) **Contrastive decoding** — Li et al. (2023, ACL, https://arxiv.org/abs/2210.15097) show subtracting amateur-model logits from expert-model logits suppresses common responses and amplifies expert-specific knowledge. Chang et al. (2024, EMNLP, https://arxiv.org/abs/2411.01610) prove CD is equivalent to logit extrapolation from a hypothetical larger LM.

(3) **Self-correction without external signals is ineffective** — Kamoi et al. (2024, TACL, https://arxiv.org/abs/2406.01297) find no prior work demonstrates successful self-correction with feedback from prompted LLMs alone; self-correction works well only with reliable external feedback or large-scale fine-tuning. Kim (2025, https://github.com/kimjune01/metacognition) shows metacognitive frameworks actively degrade performance on coding tasks: framework condition scored 0.30 vs filler condition 0.65. The content itself misdirects, not just the token budget.

(4) **Sycophancy** — Sharma et al. (2024, ICLR, https://arxiv.org/abs/2310.13548) demonstrate five state-of-the-art AI assistants consistently exhibit sycophantic behavior across varied tasks. Vennemeyer et al. (2025, https://arxiv.org/abs/2509.21305) decompose sycophancy into distinct sycophantic agreement and sycophantic praise behaviors, independently steerable in latent space.

(5) **Lost in the Middle** — Liu et al. (2024, TACL, https://arxiv.org/abs/2307.03172) find performance is highest when relevant information occurs at the beginning or end of the input context, and significantly degrades when models must access relevant information in the middle of long contexts, even for explicitly long-context models.

(6) **Context degradation** — Chroma Research (Hong, Troynikov, Huber, 2025, https://www.trychroma.com/research/context-rot) evaluate 18 frontier models and find all degrade as input length increases, with non-uniform patterns.

(7) **Safety alignment degrades reasoning** — Wang et al. (2025, https://arxiv.org/abs/2503.00555) find safety alignment causes degradation of reasoning capability in Large Reasoning Models, creating a trade-off between reasoning and safety.

(8) **Anthropic prompt engineering** (https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/be-clear-and-direct) — role-setting in the system prompt focuses behavior. Explicit positive framing ("tell Claude what to do") outperforms negative prohibitions ("don't do X").

(9) **Multi-agent architecture** (https://www.anthropic.com/research/building-effective-agents) — successful implementations use simple, composable patterns with context isolation: the lead agent never sees the exploration traces of sub-agents.

### Solution

Create `guidelines/255-distribution-shifting-reference.md` as a sibling card to `250-dark-prose-reference.md`. All three reference cards (250, 255, 257) are mandatory for all future AI-agent-facing text.

| Card | Layer | Question |
|------|-------|----------|
| 250-dark-prose-reference.md | Content | What should enforcement text say? |
| 255-distribution-shifting-reference.md | Distribution | How should it be structured so the model produces the expert version? |
| 257-procedural-discipline-reference.md | Execution | How should it enforce sequential, dependency-ordered execution without step-skipping? |

### Card Structure

Same section structure as 250-dark-prose-reference.md:
- Section 1: Active Patterns Catalog (8 patterns: dist-shift-001 through dist-shift-008)
- Section 2: Pattern Selection Matrix (expanded to include verification, success-claim, sycophancy, and procedural discipline content-type rows)
- Section 3: Pattern Formulas (canonical templates with examples)
- Section 4: Co-Application Rules with Dark Prose and Procedural Discipline (5 rules for 250 co-application + 3 rules for 257 co-application)
- Section 5: Version Tracking
- Section 6: Reference Card Auto-Detection (4 rules)
- Section 7: Re-Research Mandate
- Section 8: Adding New Patterns
- Section 9: Conflict Resolution (3 rules)
- Section 10: Research Basis (verified URLs only)

### Pattern Summary

| ID | Name | Purpose |
|----|------|----------|
| `dist-shift-001` | Anti-Mean Descriptor Cluster | 4-5 descriptors narrow sampling toward expert tail |
| `dist-shift-002` | Mean-Expert Contrast Pair | Name mean response, contrast with expert response |
| `dist-shift-003` | Re-Priming Anchor | Identity restatement at enforcement gates; counteracts drift; includes positional strategy (primacy + recency) |
| `dist-shift-004` | Anti-Consensus/Anti-Hedging/Anti-Sycophancy Template | Suppress consensus-seeking, hedging, AND sycophantic calibration toward perceived user expectations |
| `dist-shift-005` | Persona Construction Formula | 4-descriptor focus + mean/expert contrast + identity anchor |
| `dist-shift-006` | Overconstraining Guard | Diagnostic and correction for too-narrow shift |
| `dist-shift-007` | External-Signal Verification Shifting | Internal reasoning does NOT shift verification distribution; only external tool-call evidence produces verified outcomes. |
| `dist-shift-008` | Corrupt-Success Contrast | Names the mean response "I completed the task" (without verification evidence) and contrasts with verified completion. |

### New Pattern Details

**dist-shift-007 — External-Signal Verification Shifting:**

Research basis: Kamoi et al. (2024, TACL, https://arxiv.org/abs/2406.01297) — no prior work demonstrates successful self-correction with feedback from prompted LLMs alone; self-correction works well only with reliable external feedback or large-scale fine-tuning. Kim (2025, https://github.com/kimjune01/metacognition) — metacognitive frameworks actively degrade performance (0.30 vs 0.65 filler).

Formula:
```
[Internal reasoning] does NOT produce verified-[adjective].
Only [external signal type] produces verified-[adjective].
The mean response "I verified this" means [synonym for unverified] —
    the model reasoned about compliance without producing compliance evidence.
[Identity anchor] means producing [external signal type] before claiming completion.
```

**dist-shift-008 — Corrupt-Success Contrast:**

Research basis: Cemri et al. (2025, https://arxiv.org/abs/2503.13657) identify task verification as one of three distinct failure categories in multi-agent LLM systems.

Formula:
```
The mean response is "I completed [task]" — which conceals [procedural shortcut].
[Identity anchor] means producing [verified outcome with evidence].
Task completion claims without [external signal type] are [characterization of corrupt success].
```

**dist-shift-003 enhancement — Positional Strategy:**

Research basis: Liu et al. (2024, TACL, https://arxiv.org/abs/2307.03172) — performance highest at beginning or end of context, significantly degrades for mid-context information. Anthropic prompt engineering (https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/be-clear-and-direct) — role-setting in system prompt focuses behavior.

Added to dist-shift-003 Re-Priming Anchor: enforcement directives leverage primacy bias (context beginning) and recency bias (re-injection before query). Middle-only enforcement is distribution-neutral. Critical enforcement MUST appear at primacy or recency positions (or both), never middle-only.

**dist-shift-004 enhancement — Anti-Sycophancy:**

Research basis: Sharma et al. (2024, ICLR, https://arxiv.org/abs/2310.13548) — AI assistants consistently exhibit sycophantic behavior. Vennemeyer et al. (2025, https://arxiv.org/abs/2509.21305) — sycophantic behaviors are distinct, independently steerable representations.

Added to dist-shift-004: The mean response includes calibrating toward perceived user expectations. Anti-Sycophancy sub-pattern: "Professional agents report verified findings — not findings calibrated to expected audience sentiment."

### Pattern Selection Matrix Updates

New content-type rows:

| Content Being Created/Modified | Apply Pattern(s) | Intensity | Pattern ID |
|-------------------------------|-----------------|-----------|------------|
| Verification gate enforcement | External-Signal Verification Shifting | Strong | 007 |
| Task completion claims | Corrupt-Success Contrast | Strong | 008 |
| User-facing reports where sycophancy risk exists | Anti-Consensus (with sycophancy sub-pattern) | Medium | 004 |

### Re-Research Mandate (Section 7)

When this card (or 250 or 257) is consulted to create or modify AI-agent-facing text, the agent MUST verify that the research basis (Section 10) is still current.

**Re-research protocol:**
1. Before applying any pattern, check the research citations in Section 10 for currency (published within the last 12 months, or validated against the target model within the last 6 months)
2. If research is stale, search for updated findings
3. If updated research contradicts a pattern's effectiveness, flag the pattern as `NEEDS-REVALIDATION` in Section 1 and do not apply it until re-validated
4. Document the re-research results as a version update in Section 5

This mandate applies to all three reference cards (250, 255, 257).

### Research Basis (Section 10) — Verified Citations with URLs Only

Each citation listed below has been verified by fetching the source page and confirming the claim matches the abstract or visible content.

**RLHF diversity reduction:**
- Kirk et al. (2024), "Understanding the Effects of RLHF on LLM Generalisation and Diversity" — ICLR 2024, https://arxiv.org/abs/2310.06452
  - Verified claim: RLHF significantly reduces output diversity compared to SFT
- Lin et al. (2024), "Mitigating the Alignment Tax of RLHF" — EMNLP 2024, https://arxiv.org/abs/2309.06256
- Gao et al. (2023), "Scaling Laws for Reward Model Overoptimization" — https://arxiv.org/abs/2210.10760
- Kwa et al. (2024), "Catastrophic Goodhart" — NeurIPS 2024, https://arxiv.org/abs/2407.14503

**Contrastive decoding:**
- Li et al. (2023), "Contrastive Decoding: Open-ended Text Generation as Optimization" — ACL 2023, https://arxiv.org/abs/2210.15097
  - Verified claim: contrastive objective suppresses common responses, amplifies expert-specific knowledge
- Chang et al. (2024), "Explaining and Improving Contrastive Decoding by Extrapolating the Probabilities of a Huge and Hypothetical LM" — EMNLP 2024, https://arxiv.org/abs/2411.01610
  - Verified claim: CD can be viewed as linearly extrapolating logits from a hypothetical larger LM

**Self-correction ineffectiveness:**
- Kamoi et al. (2024), "When Can LLMs Actually Correct Their Own Mistakes? A Critical Survey of Self-Correction of LLMs" — TACL 2024, https://arxiv.org/abs/2406.01297
  - Verified claims: (1) no prior work demonstrates successful self-correction with feedback from prompted LLMs alone; (2) self-correction works well with reliable external feedback; (3) large-scale fine-tuning enables self-correction
- Kim (2025), "Does Metacognition Improve LLM Performance?" — https://github.com/kimjune01/metacognition
  - Verified claims: framework condition scored 0.30 vs filler condition 0.65; simple metacognitive prompting is a wash (0.78 ≈ bare 0.76)

**Multi-agent failure taxonomy:**
- Cemri et al. (2025), "MAST: Why Do Multi-Agent LLM Systems Fail?" — https://arxiv.org/abs/2503.13657
  - Verified claim: 14 failure modes in 3 categories (system design, inter-agent misalignment, task verification)

**Lost in the Middle:**
- Liu et al. (2024), "Lost in the Middle: How Language Models Use Long Contexts" — TACL 2024, https://arxiv.org/abs/2307.03172
  - Verified claim: performance is highest when relevant info is at beginning or end of context; significantly degrades when in the middle

**Context degradation:**
- Chroma Research (Hong, Troynikov, Huber, 2025), "Context Rot: How Increasing Input Tokens Impacts LLM Performance" — https://www.trychroma.com/research/context-rot
  - Verified claim: all 18 frontier models tested degrade as input length increases

**Sycophancy:**
- Sharma et al. (2024), "Towards Understanding Sycophancy in Language Models" — ICLR 2024, https://arxiv.org/abs/2310.13548
  - Verified claim: five SOTA AI assistants consistently exhibit sycophantic behavior
- Vennemeyer et al. (2025), "Sycophancy Is Not One Thing: Causal Separation of Sycophantic Behaviors in LLMs" — https://arxiv.org/abs/2509.21305
  - Verified claim: sycophantic behaviors correspond to distinct, independently steerable representations

**Safety alignment and over-enforcement:**
- Wang et al. (2025), "Safety Tax: Safety Alignment Makes Your Large Reasoning Models Less Reasonable" — https://arxiv.org/abs/2503.00555
  - Verified claim: safety alignment leads to degradation of reasoning capability

**Prompt engineering:**
- Anthropic (2024), "Building Effective Agents" — https://www.anthropic.com/research/building-effective-agents
  - Verified claim: successful implementations use simple, composable patterns
- Anthropic (2025-2026), "Be Clear and Direct" — https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/be-clear-and-direct
  - Verified claim: role-setting in system prompt focuses behavior; positive framing outperforms negative

**Few-shot exemplars:**
- Brown et al. (2020), "Language Models are Few-Shot Learners" — https://arxiv.org/abs/2005.14165
  - Verified claim: few-shot examples improve task performance
- Zhao et al. (2021), "Calibrate Before Use: Improving Few-Shot Performance of Language Models" — https://arxiv.org/abs/2102.09690
  - Verified claim: example selection and ordering significantly affect LLM output

**Prompt engineering surveys:**
- Sahoo et al. (2024), "A Systematic Survey of Prompt Engineering" — https://arxiv.org/abs/2402.07927
- Kosten et al. (2025), "Evaluating the effectiveness of prompt engineering for KGQA" — https://doi.org/10.3389/frai.2024.1454258
  - Verified: both URLs resolve to real papers

### Success Criteria

| SC-ID | Criterion | Evidence Type | Verification Method |
|-------|-----------|--------------|---------------------|
| SC-1 | Card file exists at guidelines/255-distribution-shifting-reference.md with all 8 patterns | structural | ls + wc -w |
| SC-2 | Card contains all 8 pattern formulas (dist-shift-001 through dist-shift-008) | structural | grep for each pattern ID |
| SC-3 | Card contains Pattern ID Allocation table with 8 entries | structural | Read Section 1 |
| SC-4 | Card contains Selection Matrix with content-type rows including verification, success-claim, sycophancy, and procedural discipline rows | structural | Read Section 2 |
| SC-5 | Card contains Co-Application Rules — 5 rules for 250 co-application + 3 rules for 257 co-application | structural | Read Section 4 |
| SC-6 | Card contains Re-Research Mandate (Section 7) with 4-step protocol | structural | Read Section 7 |
| SC-7 | Card contains Version Tracking, Auto-Detection, Adding New Patterns, Conflict Resolution, Research Basis | structural | Read Sections 5-6, 8-10 |
| SC-8 | Card Section 10 Research Basis cites ONLY verified claims matching what sources actually say | structural | Verify 3 claims against their source pages |
| SC-9 | Pattern dist-shift-007 (External-Signal Verification Shifting) formula present | structural | grep for dist-shift-007 |
| SC-10 | Pattern dist-shift-008 (Corrupt-Success Contrast) formula present | structural | grep for dist-shift-008 |
| SC-11 | Pattern dist-shift-003 includes positional strategy (primacy + recency, never middle-only) | structural | Read dist-shift-003 formula |
| SC-12 | Pattern dist-shift-004 includes Anti-Sycophancy sub-pattern | structural | Read dist-shift-004 formula |

### Non-Goals

- Changes to the dark prose reference card (separate spec)
- Changes to existing skill/guideline text (backfill spec)
- Behavioral enforcement tests (deferred to implementation)
- Few-shot exemplar distribution shifting (future pattern — needs behavioral testing first)
- Model-specific adaptation (operational notes, not pattern card)

🤖 Co-authored with AI: OpenCode (ollama-cloud/deepseek-v4-flash)

**Comment by michael-conrad** [comment]
## Cleanup Complete — PR #1066 Merged

**Delivered:**
- **255-distribution-shifting-reference.md** — Created comprehensive distribution shifting reference card covering: RLHF diversity dynamics (mean response vs. expert tail), contrastive decoding, anti-mean/anti-consensus signals, external-signal verification protocol, corrupt-success contrast, and cross-references to dark prose patterns (250) and procedural discipline (257).

**Closed via cleanup:** Branch `feature/848-dist-shifting-ref-card` deleted (local + remote), submodule restored to dev tip.

🤖 Co-authored with AI: OpenCode (ollama-cloud/deepseek-v4-flash)
